### DISCUSSION TITLE
Default UUID generation method.

### DISCUSSION THREAD

**Discussion by zim32** [apache/couchdb#3311] [Q&A] [answered]
I have couchdb installed from docker official image and I've noticed that sometimes after container restart my auto generated id's reset theirs prefix values.

F.e. I have
a2d30967cb18b98273f00c3ce59d2f44
a2d30967cb18b98273f00c3ce59d4f36
Then restart and ...
1ff8b14692736c8e38764ebe34004ad9

Maybe this is because container IP is changed?

**Comment by nickva** [comment, marked as answer]
The default uuid generation algorithm is "sequential". It generates a random prefix on each CouchDB node and keeps it as state. With each uuid, it increments a suffix at the end with a random increment. The random prefix changes when either a node is restarted or after the increment overflows .

https://github.com/apache/couchdb/blob/main/rel/overlay/etc/default.ini#L450-L453

**Reply by zim32** [reply]
So I can not rely on sorting by id because after restart it can regenerate prefix?

**Reply by nickva** [reply]
These are not guaranteed to increment if the prefix changes. There is a high chance you'll get incrementing uuids in the same response if you ask for say a batch of 1000 uuids (`GET .../_uuids?count=1000`) or, if you end up hitting the same node on every request and the node doesn't restart between calls.

Depending on your setup and use case, the [utc_id](https://docs.couchdb.org/en/3.1.1/config/misc.html#uuids-configuration) scheme may be useful, but that's still not a globally unique monotonically incrementing ID service. If that's what is needed you'd have to build it yourself as a separate [service](https://medium.com/better-programming/uuid-generation-snowflake-identifiers-unique-2aed8b1771bc).

**Reply by zim32** [reply]
Thanks you, utc_id is wat i need
